add-constraint.md

```diff
@@ -51,11 +51,66 @@ Adding the [Check constraint](check.html) requires that all of a column's values
-```
+To add the [Foreign Key](foreign-key.html) constraint to an existing column or group of columns, you must first use [`CREATE INDEX`](create-index.html) to index the column or columns that you plan to constrain. Then you can use the `ADD CONSTRAINT` statement to add the Foreign Key constraint to the column or columns.
+
+For example, let's say you have two simple tables, `orders` and `customers`:
+
+~~~ sql
+> SHOW CREATE TABLE customers;
+~~~
+
+~~~
++-----------+-------------------------------------------------+
+|   Table   |                   CreateTable                   |
++-----------+-------------------------------------------------+
+| customers | CREATE TABLE customers (␤                       |
+|           |     id INT NOT NULL,␤                           |
+|           |     "name" STRING NOT NULL,␤                    |
+|           |     address STRING NULL,␤                       |
+|           |     CONSTRAINT "primary" PRIMARY KEY (id ASC),␤ |
+|           |     FAMILY "primary" (id, "name", address)␤     |
+|           | )                                               |
++-----------+-------------------------------------------------+
+(1 row)
+~~~
+
+~~~ sql
+> SHOW CREATE TABLE orders;
+~~~
+
+~~~
++--------+-------------------------------------------------------------------------------------------------------------+
+| Table  |                                                 CreateTable                                                 |
++--------+-------------------------------------------------------------------------------------------------------------+
+| orders | CREATE TABLE orders (␤                                                                                      |
+|        |     id INT NOT NULL,␤                                                                                       |
+|        |     customer_id INT NULL,␤                                                                                  |
+|        |     status STRING NOT NULL,␤                                                                                |
+|        |     CONSTRAINT "primary" PRIMARY KEY (id ASC),␤                                                             |
+|        |     FAMILY "primary" (id, customer_id, status),␤                                                            |
+|        |     CONSTRAINT check_status CHECK (status IN ('open':::STRING, 'complete':::STRING, 'cancelled':::STRING))␤ |
+|        | )                                                                                                           |
++--------+-------------------------------------------------------------------------------------------------------------+
+(1 row)
+~~~
+
+To ensure that each value in the `orders.customer_id` column matches a unique value in the `orders.id` column, you want to add the Foreign Key constraint to `orders.customer_id`. So you first create an index on `orders.customer_id`:
```

Comment:
We could explain somewhere that deleting a customer would require finding any orders for that customer -- if you _already had an index on `customer_id`_ -- e.g. because your app sometimes needs to find all the orders for the customer's account page-- then that index could also be used by the FK.